libkbfs/disk_block_cache.go

```diff
@@ -0,0 +1,242 @@
+package libkbfs
+
+import (
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/keybase/client/go/logger"
+	"github.com/keybase/kbfs/kbfsblock"
+	"github.com/keybase/kbfs/kbfscrypto"
+	"github.com/keybase/kbfs/tlf"
+	"github.com/syndtr/goleveldb/leveldb"
+	"github.com/syndtr/goleveldb/leveldb/util"
+	"golang.org/x/net/context"
+)
+
+const (
+	defaultDiskBlockCacheMaxBytes uint64 = 10 * (1 << 30)
+	evictionConsiderationFactor   uint   = 3
+	blockDbFilename               string = "diskCacheBlocks.leveldb"
+	lruDbFilename                 string = "diskCacheLRU.leveldb"
+)
+
+type diskBlockCacheEntry struct {
+	Buf        []byte
+	ServerHalf kbfscrypto.BlockCryptKeyServerHalf
+}
+
+type diskBlockCacheConfig interface {
+	codecGetter
+	logMaker
+}
+
+// DiskBlockCacheStandard is the standard implementation for DiskBlockCache.
+type DiskBlockCacheStandard struct {
+	config   diskBlockCacheConfig
+	maxBytes uint64
+	log      logger.Logger
+	// protects everything below
```

Comment:
I gather from the code that the write portion of this lock is support to be protecting the writing of these fields, but not writing to the level db instances themselves.  It would be nice to clarify that in this comment.